test/integration/deployment/deployment_test.go

```diff
@@ -1379,3 +1379,121 @@ func TestDeploymentScaleSubresource(t *testing.T) {
 	// Use the scale subresource to scale the deployment down to 0
 	testScalingUsingScaleSubresource(t, tester, 0)
 }
+
+// This test verifies that the Deployment does orphan a ReplicaSet when the ReplicaSet's
+// .Labels field is changed to no longer match the Deployment's selector. It also partially
+// verifies that collision avoidance mechanism is triggered with no PodTemplateSpec change.
```

Comment:
> triggered with no PodTemplateSpec change.

triggered when a Deployment's new ReplicaSet is orphaned, even without PodTemplateSpec change.